src/servers/tests/http/influxdb_test.rs

```diff
@@ -244,3 +245,122 @@ async fn test_influxdb_write() {
+    let tx = Arc::new(tx);
+
+    let public_db_app = make_test_app(tx.clone(), None);
+    let public_db_client = TestClient::new(public_db_app);
+
+    let result = public_db_client.get("/v1/influxdb/health").send().await;
+    assert_eq!(result.status(), 200);
+
+    let result = public_db_client.get("/v1/influxdb/ping").send().await;
+    assert_eq!(result.status(), 204);
+
+    // right request with no query string
+    let result = public_db_client
+        .post("/v1/influxdb/api/v2/write")
+        .body("monitor,host=host1 cpu=1.2 1664370459457010101")
+        .header(http::header::AUTHORIZATION, "token greptime:greptime")
+        .send()
+        .await;
+    assert_eq!(result.status(), 204);
+    assert!(result.text().await.is_empty());
+
+    // right request with `bucket` query string
+    let result = public_db_client
+        .post("/v1/influxdb/api/v2/write?bucket=public")
+        .body("monitor,host=host1 cpu=1.2 1664370459457010101")
+        .header(http::header::AUTHORIZATION, "token greptime:greptime")
+        .send()
+        .await;
+    assert_eq!(result.status(), 204);
+    assert!(result.text().await.is_empty());
+
+    // right request with `db` query string
+    let result = public_db_client
+        .post("/v1/influxdb/api/v2/write?db=public")
+        .body("monitor,host=host1 cpu=1.2 1664370459457010101")
+        .header(http::header::AUTHORIZATION, "token greptime:greptime")
+        .send()
+        .await;
+    assert_eq!(result.status(), 204);
+    assert!(result.text().await.is_empty());
+
+    // make new app for 'influxdb' database
+    let app = make_test_app(tx, Some("influxdb"));
+    let client = TestClient::new(app);
+
+    // right request with `bucket` query string
+    let result = client
+        .post("/v1/influxdb/api/v2/write?bucket=influxdb")
+        .body("monitor,host=host1 cpu=1.2 1664370459457010101")
+        .header(http::header::AUTHORIZATION, "token greptime:greptime")
+        .send()
+        .await;
+    assert_eq!(result.status(), 204);
+    assert!(result.text().await.is_empty());
+
+    // right request with `db` query string
+    let result = client
+        .post("/v1/influxdb/api/v2/write?db=influxdb")
+        .body("monitor,host=host1 cpu=1.2 1664370459457010101")
+        .header(http::header::AUTHORIZATION, "token greptime:greptime")
+        .send()
+        .await;
+    assert_eq!(result.status(), 204);
+    assert!(result.text().await.is_empty());
+
+    // right request with no query string, `public_db_client` is used otherwise the auth will fail
+    let result = public_db_client
+        .post("/v1/influxdb/api/v2/write")
+        .body("monitor,host=host1 cpu=1.2 1664370459457010101")
+        .header(http::header::AUTHORIZATION, "token greptime:greptime")
+        .send()
+        .await;
+    assert_eq!(result.status(), 204);
+    assert!(result.text().await.is_empty());
+
+    // right request with the 'greptime' header and 'db' query string
```

Comment:
a couple of assertions to show that `"x-greptime-db-name"` header has a priority over the query string.